### ISSUE TITLE
Supermium v124 does not run on Windows XP SP3.

### ISSUE THREAD

**Issue by EDSln** [IDA-RE-things/Chrome-xp-api-adapter#17]
Made a dump from chrome-xpapi-adapter_x86.b5064-wD3D9accel.
https://mega.nz/file/e6ZDwDAC#MbEwzpOKfKr6EDds8Gce4izX9Pnn45LfN3Gp7ZTP3hA

**Comment by IDA-RE-things** [comment]
Ok, I have seen. The breakpoint indside my DLL reached. I think will fix this soon.

**Comment by IDA-RE-things** [comment]
Try `chrome-xpapi-adapter.5065.for.testing.zip`. Should work.
The breakpoin was on not called function.
If some another one will happen, will continue to fix.

**Comment by EDSln** [comment]
Yes, now it starts up without errors. Hardware acceleration works the same way. Thank you!

**Comment by EDSln** [comment]
I won't close the issue just yet, in case someone else has it unrepaired.

**Comment by MWF2** [comment]
![image](https://github.com/IDA-RE-things/Chrome-xp-api-adapter/assets/74994144/1e343f61-44f9-4711-9df7-d6bbafc8f032)
now a error about AcquireSRWLockExclusive

**Comment by EDSln** [comment]
Did you use 
chrome-xpapi-adapter_x86.b5064-wD3D9accel.zip and the file from
chrome-xpapi-adapter.5065.for.testing.zip? 
And what OS, including bitness and SP?

**Comment by IDA-RE-things** [comment]
@MWF2 , I hava recheked, Yes, this function somehow disappear from Vista/Win7 variant: `chrome-win7-xp-api-router-Vista.dll` 
I will prepare the fix.

**Comment by IDA-RE-things** [comment]
UPD: no, all the function here.
So, what the DLL you are using ? What OS ?

**Comment by IDA-RE-things** [comment]
I have rechecked: boths `chrome-win7-xp-api-router.dll` and `chrome-win7-xp-api-router-Vista.dll` have this functions.
So, what the DLL you are using ? What OS ?

**Comment by JoachimHenze** [comment]
I can confirm, all on XPSP3 german:

* "Supermium 122.0.6261.152 (R6)" starts up fine with its stock progwrp.dll.
* "Supermium 124.0.6367.245 prebuild with its stock progwrp.dll 1.1.0.5016 from win32ss" does **not** start up.

* "Supermium 124.0.6367.245 prebuild" with all below:
a.) D3DCompiler_XP.dll extracted from chrome-xpapi-adapter_x86.b5064-wD3D9accel.zip from https://github.com/IDA-RE-things
b.) progwrp.dll v1.2.0.5064 renamed from "chrome-win7-xp-api-router.dll" which itself extracted from "chrome-xpapi-adapter_x86.b5064-wD3D9accel.zip" from https://github.com/IDA-RE-things
c.) chrome-xpapi-adapter.dll v1.2.5065 renamed from "chrome-xpapi-adapter.5065.dll" which itself extracted from "chrome-xpapi-adapter_x86.b5064-wD3D9accel.zip" from https://github.com/IDA-RE-things
**This combination works for me.**

It is getting a bit confusing to setup and rename all of that, but it works for me.

**Comment by IDA-RE-things** [comment]
now I have updated the distributive and all requered DLL's included in main archive.
Separate 5065 dll was only for testing on machines, having problem with previous build.

**Comment by JoachimHenze** [comment]
Yes, and I do confirm that your "All-in-one" chrome-xpapi-adapter_x86.b5065-wD3D9accel.zip does work very well for me in combination with "Supermium 124.0.6367.245 prebuild" on XPSP3. I'd say the stability **on XPSP3** is similar for me to what I had before with
"Supermium 122.0.6261.152 (R6) + your "chrome-xpapi-adapter_x86.b5061.2-wGPUaccel.zip".
GPU accel is also working fine (aside the already known https://github.com/IDA-RE-things/Chrome-xp-api-adapter/issues/15 )

What I do appreciate most in the new combination is that 124_pre is slightly smaller than R122_R6 and that the `--ungoogled-supermium` flag is a bit superior over the old version.
**I do perceive it as an upgrade without regret on XPSP3 x86.**

I am not so enthusiastic yet about the performance of 124_pre x86 without any of IDA-RE-stuff on Win8.1x64 though.
But that is not your job to improve on that.

**Comment by IDA-RE-things** [comment]
> I am not so enthusiastic yet about the performance of 124_pre x86 without any of IDA-RE-stuff on Win8.1x64 though.

@JoachimHenze , So you are not want to use my work and will be happy if it will work without my DLLs' ? I'm understand this right?

**Comment by JoachimHenze** [comment]
I guess we ran into some sort of language barrier here.
To understand me right: I do love your work, I do love your dlls, and I do use your dlls on XPSP3 always, where they are vastly superior over the stock-shipped progwrp.dll from win32ss.


On another PC with Win8.1x64 I didn't ever use your dlls yet. Because I didn't expect them to be necessary there. Other OS.
There I do only install the Supermium without any further modifications to its binaries. In this case I do still suffer from some crashes, but since your work is not involved there, there is also nothing for you to fix in that case. Your dlls are not to be blamed. Do your dlls improve anything on Win8.1x64 too?

**Comment by IDA-RE-things** [comment]
Ok, thanks for response. I'm understand that you mean another computer with different OS.
No, my DLL's do nothing with Win7+ systems, because no need to emulate any API for them (while its such).
But question was initiated not by your posts above, but activity on Supermium issues tracker. What I mean:

When you comparing there (and not only you) our two DLLs, and even in detail by features; say, something is missing or not working.

Yes, I am certainly pleased to know that my DLL is superior to the original and to read such a review. But there is another side of the coin.

Because his ambitions and the need to fight for an audience push him to improve his DLL (and does not matter already: wich way -- silently taking ideas from my DLL or not; instead of agreeing that users should use a third-party DLL for XP at least).

But its not fine for me and I not happy from this.

He doesn't want to be a team, a corporation (as he called himself) or anything else. He wants to be alone. (what is it worth his impudence to name himself personally the Microsoft brand - the Win32 subsystem in his 19 yo. and now 22. You can afford such a nickname somewhere in a small chat among people who write in javascript or are not programmers at all in order to somehow distinguish himself from them with such name. But not among system programmers. (BTW, what will the guys from Microsoft itself say about this?). And such person (with such ambitions) will also not help the ReatOS project. Because they are  competitors as fact.
I already wrote this in one of the posts on the Supermium tracker.

It may be nothing. But why should I see his nickname every time in my VisualStudio (configuration name) when working with projects or in directory names?
It is personifying oneself with something that one is not.
With same success, he could call himself other companies trademarks (appropriate them) : like guy "MS World", or take a name of my car name in case of good service-station-man?. (WTF?)

...

Resuming, I also have my own interests and my own motivation.
As soon as his DLL reached the same features as my DLL and users say “no differences, then we don’t need a third-party DLL",
this will push me to take the next step - to make it Open Source (as I originally intended).
Probably many 3rd persons will be happy from this. But not the author of the original, who planned to build a far-reaching business based on his DLL.
...

And I’ll say it again - I’m sorry that it turned out that we are not the one collaborated team (as it in ReatOS as example, or others), but separate competitors.

**Comment by EDSln** [comment]
It's certainly bad enough that you are competitors rather than partners. If your efforts with win32ss were combined, the result would be even more impressive than it is now. But on the other hand, competition also forces everyone to do a better job. So in any case, thank you very much for your work, for the fact that there is another implementation.
